src/helper.cc

```diff
@@ -474,14 +474,12 @@ helper::submitRequest(Helper::Xaction *r)
 static void
 SubmissionFailure(const helper::Pointer &hlp, HLPCB *callback, void *data)
 {
-    auto result = Helper::Error;
     if (!hlp) {
         debugs(84, 3, "no helper");
-        result = Helper::Unknown;
+        callback(data, Helper::Reply(Helper::Unknown));
+    } else { // pretend the helper has responded with ERR
+        hlp->callBack(callback, data, Helper::Reply(Helper::Error));
```

Comment:
Please do not send answers to _never-submitted_ queries through the helper. Keep the old direct callback-calling code virtually unchanged here.

Reply:
Left this code as-is in PR229..